web_monitoring/cli.py

```diff
@@ -0,0 +1,67 @@
+# Command Line Interface
+# See scripts/ directory for associated executable(s). All of the interesting
+# functionality is implemented in this module to make it easier to test.
+from docopt import docopt
+from tqdm import tqdm
+from web_monitoring import internetarchive as ia
+from web_monitoring import pf_edgi as pf
+from web_monitoring import db
+
+
+# These functions lump together library code into monolithic operations for the
+# CLI. They also print. To access this functionality programmatically, it is
+# better to use the underlying library code.
+
+
+def import_ia(url, agency, site):
+    print('obtaining versions list from Internet Archive...')
+    versions = ia.list_versions(url)
+    # Collect all the results and POST them as a unit (rather than streaming).
+    formatted_versions = []
+    for dt, uri in tqdm(versions, desc='formatting', unit='versions'):
+        version = ia.timestamped_uri_to_version(dt, uri,
+                                                url=url,
+                                                site=site,
+                                                agency=agency)
+        formatted_versions.append(version)
+    print('posting to db....')
+    res = db.post_versions(formatted_versions)
+    assert res.ok
+    print(res.json())
+
+
+def import_pf_archive(cabinet_id, archive_id, *, agency, site):
```

Comment:
Fortunately, it doesn't soak up anything. If you want it to soak up extra position args, you have to put a variable beside it like `*args`. What is does is force all args after it to be called with the keywords: it makes `agency` and `site` "keyword-only arguments." For functions with many arguments and/or arguments with no obvious ordering, it's a great way to avoid bugs. It also means that you can re-order these arguments in the future without breaking the API.